### PR TITLE
Optionally set github personal access token in plugin config

### PR DESCRIPTION
**Pull request by ggirodda**
Hi, thank you for the strapi plugin, it is really useful if we don't want to trigger the github action for every change, when uising web hooks.
I made same changes
- POST method instead of GET for triggering route, considering that there'll be an axios post call to github, that seemed to me more appropriate
- update the axios version, because of the npm critical alert
- added the possibility to set the personal access token using plugin config, if someone doesn't want his / her token to be visible / editable from the workflow form

I added also some test on the added code.

Thanks !

PS. II think that you should set at least the link to github repository and your github username in package.json, because there's no information for now on npm website about the package

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -19,6 +19,28 @@ First, you need to add one or multiple `workflows` in your content.
 - Add a new workflow by filling all its details
 - Once saved, you can trigger the workflow by going to Plugins > GitHub Actions Plugin and clicking on the 'start' button.
 
+### Set personal access token in config
+If you don't want your personal access token to be visible and editable from the workflow,
+you can set it in the plugin config, using for example a env variable
+
+![Alt text](/strapi-workflow-example.png?raw=true "Strapi Workflow example")
```
Comment (lodi-g) [inline README.md:26]: Could you please move the image to a docs folder?

File: services/github-actions.js [hunk 2]
```diff
@@ -1,7 +1,35 @@
...
+
+    const config = utils.getConfig()
+    const hasEnvPat = config.hasEnvPat
+
+    if (!hasEnvPat) {
+      return workflow
+    }
+    const wfConfigPat = config.pats[workflow.name]
+
+    if (wfConfigPat) {
```
Comment (lodi-g) [inline services/github-actions.js:18]: Could you log some warning if hasEnvPat is set to true but `config.pats[workflow.name]` doesn't exist. That way we can warn about typos in the config file.

**Review by lodi-g — CHANGES_REQUESTED** [review]
Thank you for the quality addition! I have just left 2 comments if you can address them before I merge that'd be great.
I'll also add the repo information for the next release, thanks for the heads up.

**Comment by lodi-g** [comment]
Merging in #2 with requested changes

**Comment by lodi-g** [comment]
Published on npm with version 0.1.0

**Comment by ggirodda** [comment]
Great, thank you! Sorry for the late reply

**Pull request closed without merging**